index.php

```diff
@@ -36,48 +42,81 @@ function displayAllRecords(array $records): string
     return $htmlOutput;
 }
 
-// Handle add record
+//display genres function
+function displayAllGenres(array $genres):string
+{
+    $htmlOutput = '';
+
+    foreach ($genres as $genre){
+        $htmlOutput .=
+        "<option value={$genre['id']}>{$genre['name']}</option>";
+    }
+
+    return  $htmlOutput;
+}
+
+
+// Handle add-record input
 $newAlbumName = $_POST['newAlbumName'] ?? false;
 $newArtistName = $_POST['newArtistName'] ?? false;
 $newReleaseYear = $_POST['newReleaseYear'] ?? false;
 $newGenre = $_POST['newGenre'] ?? false;
 $newScore = $_POST['newScore'] ?? false;
 $newImg = $_POST['newImg'] ?? false;
 
-$diplayFormErrors = false;
-// $displayFormSucces = false;
-// $formSuccess = 'record added to collection';
+//Generate form submit errors function 
+function generateFormSubmitErrors(
+    $newAlbumName,
+    $newArtistName,
+    $newReleaseYear,
+    $newGenre,
+    $newScore,
+    $newImg
+):array
+{
+        $errors = [];
 
-//on submit...
-if (isset($_POST['newRecord'])) {
+        if (empty($newAlbumName)) {
+            $errors['albumName'] = 'Album name is required';
+        }
+        if (empty($newArtistName)) {
+            $errors['artistName'] = 'Artist name is required';
+        }
+        if (empty($newReleaseYear) || !is_numeric($newReleaseYear) || strlen((string)$newReleaseYear) != 4) {
+            $errors['releaseYear'] = 'Invalid release year';
+        }
+        if ($newGenre === 0 || empty($newGenre)) {
+            $errors['genre'] = 'Music genre is required';
+        }
+        if (empty($newScore) || !is_numeric($newScore) || $newScore < 1 || $newScore > 10) {
+            $errors['score'] = 'Number between 1 and 10';
+        }
+        if (empty($newImg) || !preg_match('/\bhttps?:\/\/\S+\.(jpg)\b/i', $newImg)) {
+            $errors['img'] = 'Image link is required';
+        }
 
-    //handle errors
-    $errors = [];
+        return $errors;
```

Comment:
IIt looks like this chunk is indented too far

You should also move this function into its own file